pkg/git/service.go

```diff
@@ -177,17 +177,61 @@ func (s *Service) SetGitConfig(userData *gitprovider.GitUser) error {
 		}
 	}
 
+	if err := setSigningConfig(cfg, providerConfig); err != nil {
+		return err
+	}
+
 	var buf bytes.Buffer
 	_, err = cfg.WriteTo(&buf)
 	if err != nil {
 		return err
 	}
 
-	err = os.WriteFile(gitConfigFileName, buf.Bytes(), 0644)
+	return os.WriteFile(gitConfigFileName, buf.Bytes(), 0644)
+}
+
+func setSigningConfig(cfg *ini.File, providerConfig *gitprovider.GitProviderConfig) error {
+	if providerConfig == nil || providerConfig.SigningMethod == nil || providerConfig.SigningKey == nil {
+		return nil
+	}
+
+	if !cfg.HasSection("user") {
+		_, err := cfg.NewSection("user")
+		if err != nil {
+			return err
+		}
+	}
+
+	_, err := cfg.Section("user").NewKey("signingkey", *providerConfig.SigningKey)
 	if err != nil {
 		return err
 	}
 
+	if !cfg.HasSection("commit") {
+		_, err := cfg.NewSection("commit")
+		if err != nil {
+			return err
+		}
+	}
+
+	switch *providerConfig.SigningMethod {
+	case gitprovider.SigningMethodGPG:
+		_, err := cfg.Section("commit").NewKey("gpgSign", "true")
+		if err != nil {
+			return err
+		}
+	case gitprovider.SigningMethodSSH:
+		if !cfg.HasSection("gpg") {
```

Comment:
It would be great if we auto-configure signature verification as well.

You can check what needs to be done here https://docs.gitlab.com/ee/user/project/repository/signed_commits/ssh.html#verify-commits-locally.

Reply:
Done.